test-data/unit/check-python310.test

```diff
@@ -1220,6 +1220,21 @@ def main() -> None:
         case a:
             reveal_type(a)  # N: Revealed type is "builtins.int"
 
+[case testMatchCapturePatternFromAsyncFunctionReturningUnion-xfail]
+async def func1(arg: bool) -> str | int: ...
+async def func2(arg: bool) -> bytes | int: ...
+
+async def main() -> None:
+    match await func1(True):
+        case str(a):
+            match await func2(True):
```

Comment:
Does this nesting add anything to the test? Better to keep tests simple unless the additional complexity tests something interesting.

Reply:
I was following the lead of the preceding test case. See above ^